client/src/main/java/io/confluent/kafka/schemaregistry/client/CachedSchemaRegistryClient.java

```diff
@@ -233,7 +267,21 @@ private int registerAndGetId(
 
   protected ParsedSchema getSchemaByIdFromRegistry(int id, String subject)
       throws IOException, RestClientException {
-    SchemaString restSchema = restService.getId(id, subject);
+    if (missingIdCache.getIfPresent(new SubjectAndId(subject, id)) != null) {
+      throw new RestClientException("This ID is banned",
```

Comment:
This error message should be changed to be the same as the error message in the response from the server.